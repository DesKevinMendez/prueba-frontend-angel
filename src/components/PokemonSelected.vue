<template>

    <div 
    id="container" 
    class="mx-auto w-2/5 bg-blue-100 mb-4 cursor-pointer"
    @click="$router.push({ name:'pokemon', params: { id:poke.id}})"
    >
        <h2 class="text-left text-blue-600 border-2 border-blue-600 h-10 pt-2 pb-3 px-3 font-bold">{{ poke.name }}</h2>
    </div>
    
</template>


<script>

export default {
    props:{
        poke:{
            type: Object,
            required: true
        }
    },
}


</script>

