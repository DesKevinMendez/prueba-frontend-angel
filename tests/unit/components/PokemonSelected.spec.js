import { shallowMount } from "@vue/test-utils"
import PokemonSelected from '@/components/PokemonSelected.vue'
import { pokemonState } from "../mock-data/test-pokemon-state";



describe( "Component: PokemonSelected", () => {

    const mockRouter = {
        push: jest.fn()
    }

    const wrapper = shallowMount(PokemonSelected, {

        props: {
            poke: pokemonState.pokemons[1]
        },
        global: {
            mocks:{
                $router: mockRouter
            }
        }

    })

    test( "Debe de hacer match con el snapshot", () => {

        expect(wrapper.html()).toMatchSnapshot()

    })

    test( "Debe de redireccionar al hacer click en un elemento", () => {

        const element = wrapper.find('.cursor-pointer')
        console.log("Pokemon seleccionado", element)
        element.trigger('click')

        expect(mockRouter.push).toHaveBeenCalled()
    })

} )