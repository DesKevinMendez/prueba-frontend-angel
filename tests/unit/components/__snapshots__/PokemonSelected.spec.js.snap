// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: PokemonSelected Debe de hacer match con el snapshot 1`] = `
<div id="container" class="mx-auto w-2/5 bg-blue-100 mb-4 cursor-pointer">
  <h2 class="text-left text-blue-600 border-2 border-green-600 h-15 pt-5 pb-5 px-5 font-bold">ivysaur</h2>
</div>
`;
